examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts

```diff
@@ -136,3 +235,151 @@ export function addLanguage(patient: Patient, valueCoding: Coding, preferred: bo
 
   patient.communication = patientCommunications;
 }
+
+/**
+ * Adds a Coverage resource
+ *
+ * @param medplum - The Medplum client
+ * @param patient - The patient beneficiary of the coverage
+ * @param answers - A list of objects where the keys are the linkIds of the fields used to set up a
+ *                  coverage (see getGroupRepeatedAnswers)
+ */
+export async function addCoverage(
+  medplum: MedplumClient,
+  patient: Patient,
+  answers: Record<string, QuestionnaireResponseItemAnswer>
+): Promise<void> {
+  const payor = answers['insurance-provider'].valueReference as Reference<Organization>;
+  const subscriberId = answers['subscriber-id'].valueString;
+
+  await medplum.upsertResource(
+    {
+      resourceType: 'Coverage',
+      status: 'active',
+      beneficiary: createReference(patient),
+      subscriberId: subscriberId,
+      relationship: { coding: [answers['relationship-to-subscriber'].valueCoding as Coding] },
+      payor: [payor],
+    },
+    {
+      beneficiary: getReferenceString(patient),
+      payor: getReferenceString(payor),
+      // subscriberId: subscriberId,
```

Comment:
This is unexpected to me but including `subscriberId` in the query makes the test fail. Not sure if a problem with the test client or this is expected behavior

Reply:
What is the error message?

Reply:
Not an error, [the test just breaks](https://github.com/medplum/medplum/pull/4851/files#diff-fe5ca34e17f32d220c942739b3d9c6ecc4250bc22cc37d8db74dc94616723bf4R241), the upsert doesn't work and the object is created twice when it shouldn't.

Reply:
Can you file a separate issue for this? Let's debug in a separate PR

Reply:
Here is the link to the issue: https://github.com/medplum/medplum/issues/4957